### PR TITLE
Removal of IBC infrastructure

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This represents a course removal of the IBC infrastructure.

Questions on what else to remove can start with looking at the following under `src/coreclr`.
```
szZapBBInstr
szZapBBInstrDir
void Module::CreateProfilingData();
void Module::DeleteProfilingData();
```

Files:
`src/coreclr/inc/corbbtprof.h`
Under `src/coreclr/tools/aot/ILCompiler.ReadyToRun/IBC`

/cc @davidwrighton

### CODE REVIEW

File: src/coreclr/vm/clsload.cpp [hunk 1]
```diff
@@ -951,17 +951,6 @@ TypeHandle ClassLoader::LoadConstructedTypeThrowing(TypeKey *pKey,
         typeHnd = pKey->GetModule()->LookupFullyCanonicalInstantiation(pKey->GetTypeToken(), &existingLoadLevel);
     }
 
-    if (typeHnd.IsNull())
-    {
-        typeHnd = LookupTypeHandleForTypeKey(pKey);
```
Comment (davidwrighton) [inline src/coreclr/vm/clsload.cpp:956]: This stuff shouldn't be removed. Only the call to the logger should go.

File: src/coreclr/vm/ceeload.cpp [hunk 2]
```diff
@@ -5860,556 +5766,6 @@ void SaveManagedCommandLine(LPCWSTR pwzAssemblyPath, int argc, LPCWSTR *argv)
 #endif
 }
```
Comment (davidwrighton) [inline src/coreclr/vm/ceeload.cpp:5768]: You should also be able to delete the ProfileEmitter class

Comment (davidwrighton) [reply src/coreclr/vm/ceeload.cpp:5768]: As well as the ProfileMap class

File: src/coreclr/vm/ceeload.cpp [hunk 3]
```diff
@@ -6512,7 +5868,6 @@ void Module::LogTokenAccess(mdToken token, SectionFormat format, ULONG flagnum)
         NOTHROW;
```
Comment (davidwrighton) [inline src/coreclr/vm/ceeload.cpp:5868]: You should be able to delete the LogTokenAccess function

File: src/coreclr/vm/ceeload.cpp [hunk 4]
```diff
@@ -6783,8 +6138,7 @@ idTypeSpec Module::LogInstantiatedType(TypeHandle typeHnd, ULONG flagNum)
     {
         NOTHROW;
         GC_NOTRIGGER;
-        MODE_ANY;
```
Comment (davidwrighton) [inline src/coreclr/vm/ceeload.cpp:6786]: You should be able to delete LogInstantiatedType and LogInstantiatedMethod. As well as TokenDefinitionHelper and Module::EncodeModuleHelper.

File: src/coreclr/vm/zapsig.cpp [hunk 5]
```diff
@@ -233,13 +233,6 @@ BOOL ZapSig::GetSignatureForTypeHandle(TypeHandle      handle,
 
     if ((index != 0) && (this->pfnTokenDefinition != NULL))
     {
-        //
-        // We do not want to log the metadata lookups that we perform here
-        //
-        IBCLoggingDisabler   disableLogging;
-
```
Comment (davidwrighton) [inline src/coreclr/vm/zapsig.cpp:240]: Please delete ZapSig::ExternalTokens::IbcTokens and logic that uses it.

File: src/coreclr/vm/ceeload.h [hunk 6]
```diff
@@ -1428,8 +1428,6 @@ class Module
 
         BAD_FORMAT_NOTHROW_ASSERT(TypeFromToken(token) == mdtTypeDef);
```
Comment (davidwrighton) [inline src/coreclr/vm/ceeload.h:1430]: The ProfilingBlobEntry type and all its derivatives should go. IbcNameHandle should go. 
ProfilingBlobTable should go. TokenProfileData should go. 

m_nativeImageProfiling should go
m_methodProfileList should be deleted

CORCOMPILE_METHOD_PROFILE_LIST should go

Delete corbbtprof.h

CorProfileData should go.

Remove the HOT_MODEL from metadata

**Review by davidwrighton — CHANGES_REQUESTED** [review]
Good progress. I found 1 thing that shouldn't have been removed which is quite important (and will cause perf problems if you don't fix), but I also found a bunch of more stuff to delete.

File: src/coreclr/vm/clsload.cpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/clsload.cpp:956]: Ugh. I was stuck on this for about 10 minutes trying to determine if I should delete this branch. Thanks for confirming.

File: src/coreclr/vm/ceeload.h (hunk 6 above)
Comment (davidwrighton) [reply src/coreclr/vm/ceeload.h:1132]: We should just delete the entire ProfileDataNode.cs file, its tied into the /tuning switch to crossgen2.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/ceeload.h:1132]: Done.

**Comment by davidwrighton** [comment]
We also need to delete the /tuning switch, which exists to make use of the ProfileDataNode stuff you just deleted.

crossgen2\CommandLineOptions.cs:        public bool Tuning;
crossgen2\CommandLineOptions.cs:                syntax.DefineOption("tuning", ref Tuning, SR.TuningImageOption);
crossgen2\Program.cs:                        .UseIbcTuning(_commandLineOptions.Tuning)
crossgen2\Properties\Resources.resx:  <data name="TuningImageOption" xml:space="preserve">
ILCompiler.ReadyToRun\Compiler\ReadyToRunCodegenCompilationBuilder.cs:        private bool _ibcTuning;
ILCompiler.ReadyToRun\Compiler\ReadyToRunCodegenCompilationBuilder.cs:        public ReadyToRunCodegenCompilationBuilder UseIbcTuning(bool ibcTuning)
ILCompiler.ReadyToRun\Compiler\ReadyToRunCodegenCompilationBuilder.cs:            _ibcTuning = ibcTuning;
ILCompiler.ReadyToRun\Compiler\ReadyToRunCodegenCompilationBuilder.cs:            if (_ibcTuning)

All of that should go.

---
In reply to: [958156947](https://github.com/dotnet/runtime/pull/68717#issuecomment-958156947) [](http://example.com/codeflow?ancestors=958156947,958156947)

**Review by davidwrighton — APPROVED** [review]

**Pull request merged**